client/question/services/ConversationManagerServiceTestUtils.js

```diff
@@ -79,38 +70,30 @@ tie.factory('ExpectedFeedbackObjectFactory', [function() {
    */
   ExpectedFeedback.prototype.verifyFeedback = function(submissionResult) {
     var errorMessages = [];
+    var observedFeedback = submissionResult.getFeedback();
+    var observedFeedbackParagraphObject = observedFeedback.getParagraphs();
+    var observedStdout = submissionResult.getStdout();
+    var expectedStdout = this._stdout;
 
-    var feedback = submissionResult.getFeedback();
-
-    if (this._firstFeedbackParagraphContent !== null) {
-      var observedContent = feedback.getParagraphs()[0].getContent();
-      var expectedContent = this._firstFeedbackParagraphContent;
-      if (observedContent !== expectedContent) {
-        errorMessages.push(
-          'Bad content for first feedback paragraph: expected "' +
-          expectedContent + '" but received "' + observedContent + '"');
-      }
-    }
-
-    if (this._secondFeedbackParagraphContent !== null) {
-      observedContent = feedback.getParagraphs()[1].getContent();
-      expectedContent = this._secondFeedbackParagraphContent;
-      if (observedContent !== expectedContent) {
-        errorMessages.push(
-          'Bad content for second feedback paragraph: expected "' +
-          expectedContent + '" but received "' + observedContent + '"');
+    for (var i in this._expectedFeedbackParagraphs) {
```

Comment:
Seems a bit odd to me that this works when this._expectedFeedbackParagraphs is null, btw (hence the comment elsewhere about making it []).

Reply:
This is because <code>for (var i in null)</code> is valid but <code>for (var i = 0; i < null.length; i++)</code> is not